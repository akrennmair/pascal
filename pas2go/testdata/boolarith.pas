program test;

var a, b : boolean;

begin
	a := true;
	b := false;
	writeln('boolean succ(true): ', succ(a));
	writeln('boolean succ(false): ', succ(b));
	writeln('boolean succ(succ(false)): ', succ(succ(b)));
	writeln('boolean pred(true): ', pred(a));
	writeln('boolean pred(false): ', pred(b));
	writeln('boolean pred(pred(false)): ', pred(pred(b)));
end.
