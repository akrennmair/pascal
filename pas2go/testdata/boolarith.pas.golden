package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	var (
		a bool
		b bool
	)
	_ = a
	_ = b

	a = true
	b = false
	system.Writeln("boolean succ(true): ", system.BoolSucc(a))
	system.Writeln("boolean succ(false): ", system.BoolSucc(b))
	system.Writeln("boolean succ(succ(false)): ", system.BoolSucc(system.BoolSucc(b)))
	system.Writeln("boolean pred(true): ", system.BoolPred(a))
	system.Writeln("boolean pred(false): ", system.BoolPred(b))
	system.Writeln("boolean pred(pred(false)): ", system.BoolPred(system.BoolPred(b)))
}
